pkg/algorithms/cmr/cmr.py

```diff
@@ -0,0 +1,123 @@
+"""EDivisive Algorithm from hunter"""
+
+# pylint: disable = line-too-long
+from typing import List
+import pandas as pd
+import numpy
+
+from fmatch.logrus import SingletonLogger
+from hunter.series import  ChangePoint, ComparativeStats
+from pkg.algorithms.algorithm import Algorithm
+
+
+class CMR(Algorithm):
+    """Implementation of the CMR algorithm
+    Will Combine metrics into 2 lines and compare with a tolerancy to set pass fail
+
+    Args:
+        Algorithm (Algorithm): Inherits
+    """
+
+
+    def _analyze(self):
+        """Analyze the dataframe with meaning any previous data and generate percent change with a current uuid
+
+        Returns:
+            series: data series that contains attributes and full dataframe
+            change_points_by_metric: list of ChangePoints
+        """
+        logger_instance = SingletonLogger.getLogger("Orion")
+        logger_instance.info("Starting analysis using Isolation Forest")
+        self.dataframe["timestamp"] = pd.to_datetime(self.dataframe["timestamp"])
+        self.dataframe["timestamp"] = self.dataframe["timestamp"].astype(int) // 10**9
+
+        logger_instance.info('data frame ' + str(self.dataframe))
+
+        # if larger than 2 rows, need to get the mean of 0 through -2
+        self.dataframe = self.combine_data_frames( self.dataframe)
+
+        series= self.setup_series()
+        tolerancy = 20
+        metric_columns = self.metrics_config.keys()
+        df, change_points_by_metric = self.run_cmr(tolerancy, metric_columns, self.dataframe)
+        series.data= df
+        return series, change_points_by_metric
+
+
+    def run_cmr(self, tolerancy: int,metric_columns: List[str],  dataframe_list: pd.DataFrame):
+        """
+        Generate the percent difference in a 2 row dataframe
+
+        Args:
+            tolerancy (int): tolerancy to compare on 
+            metric_columns (List[str]): string list of metric column names
+            dataframe_list (pd.DataFrame): data frame of all data to compare on
+
+        Returns:
+            pd.Dataframe, dict[metric_name, ChangePoint]: Returned data frame and change points
+        """
+        change_points_by_metric={ k:[] for k in metric_columns }
+        max_date_time = pd.Timestamp.max.to_pydatetime()
+        max_time = max_date_time.timestamp()
+        difference = ["difference", max_time]
```

Comment:
difference and pass_fail can be omitted and difference[1] below can be substituted using max_time or timestamp of previous row.